src/js/components/Select/stories/LazyLoading.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component, PureComponent } from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { Box, CheckBox, Grommet, Select } from 'grommet';
+import { grommet } from 'grommet/themes';
+
+const dummyOptions = Array(2000)
+  .fill()
+  .map((_, i) => `option ${i}`)
+  .sort((a, b) =>
+    a.localeCompare(b, undefined, { numeric: true, sensitivity: 'base' }),
+  );
+
+class Option extends PureComponent {
+  render() {
+    const { value, selected } = this.props;
+    return (
+      <Box direction="row" gap="small" align="center" pad="xsmall">
+        <CheckBox tabIndex="-1" checked={selected} onChange={() => {}} />
+        {value}
+      </Box>
+    );
+  }
+}
+
+class LazyLoading extends Component {
+  state = {
+    selected: [],
+    options: dummyOptions.slice(0, 200),
+  };
+
+  onMore = () => {
+    setTimeout(() => {
+      const { options } = this.state;
+      console.log('onmore called');
+      this.setState({ options: dummyOptions.slice(0, options.length + 200) });
+    }, 1000);
+  };
+
+  onClose = () => {
+    const { selected, options } = this.state;
+    this.setState({
+      options: options.sort((p1, p2) => {
```

Comment:
Why do we need to sort the options when the Select drop down is closed?

Reply:
Maybe because of replace?
It was in manyoptions.js. I wasn't sure so didn't remove.

Reply:
I think it's for moving selected options to the top.
But its not working in both this one and manyoptions.js